scheduler/src/cook/kubernetes/api.clj

```diff
@@ -1702,7 +1713,14 @@
     (.setRestartPolicy pod-spec "Never")
 
     (.addTolerationsItem pod-spec toleration-for-deletion-candidate-of-autoscaler)
-    (.addTolerationsItem pod-spec (toleration-for-pool cook-pool-taint-name cook-pool-taint-prefix pool-name))
+    ; Cook supports two kinds of tolerations on nodes.
+    ; 1. It can tolerate a node, where the value of that toleration depends on the pool name.
+    ; I.e.,  <KEY1>=<PREFIX><POOLNAME>
+    (when cook-pool-taint-name
+      (.addTolerationsItem pod-spec (toleration-for-pool cook-pool-taint-name cook-pool-taint-prefix pool-name)))
+    ; 2. Or it can statically tolerate a particular KEY2 and VAL2.
+    (when cook-pool-taint2-name
+      (.addTolerationsItem pod-spec (toleration-for-pool2 cook-pool-taint2-name cook-pool-taint2-value)))
```

Comment:
Ah, here's where it 1 & 2 come together. 
I don't love this. Specifically, I don't love that 1 & 2 are logically different and that difference isn't reflected in their config keys. I was assuming that 1 & 2 were applied as `1 || 2` or `1 && 2` and was ready to comment about that. I didn't realize that 1 is a prefix and 2 is not.

Reply:
Oh, I stand corrected. It's there as `cook-pool-taint-prefix` vs `cook-pool-taint2-name`